### PR TITLE
Synonyms bugfix - prevent index not found when system index has not been created

### PR DESCRIPTION
**Pull request by carlosdelest**
When `DELETE _synonyms/synonym-set/synonym-rule` is executed and the system index does not yet exist, the response is:

```json
{
  "error": {
    "root_cause": [
      {
        "type": "index_not_found_exception",
        "reason": "no such index [.synonyms]",
        "resource.type": "index_or_alias",
        "resource.id": ".synonyms",
        "index_uuid": "_na_",
        "index": ".synonyms"
      }
    ],
    "type": "index_not_found_exception",
    "reason": "no such index [.synonyms]",
    "resource.type": "index_or_alias",
    "resource.id": ".synonyms",
    "index_uuid": "_na_",
    "index": ".synonyms"
  },
  "status": 404
}
```

This PR returns a resource not found for synonyms set instead, adds tests for it, and makes error messages more homogeneous in terms of casing.

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml [hunk 1]
```diff
@@ -5,9 +5,6 @@ setup:
 
 ---
 "Create synonyms with no ID creates ID automatically":
-  - skip:
```
Comment (kderusso) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml:8]: Was this intentionally removed?

File: server/src/main/java/org/elasticsearch/action/synonyms/PutSynonymRuleAction.java [hunk 2]
```diff
@@ -60,10 +61,11 @@ public Request(StreamInput in) throws IOException {
...
         public Request(String synonymsSetId, String synonymRuleId, BytesReference content, XContentType contentType) throws IOException {
             this.synonymsSetId = synonymsSetId;
-            this.synonymRule = PARSER.apply(
-                XContentHelper.createParser(XContentParserConfiguration.EMPTY, content, contentType),
-                synonymRuleId
-            );
+            try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, content, contentType)) {
+                this.synonymRule = PARSER.apply(parser, synonymRuleId);
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Failed to parse: " + content.utf8ToString(), e);
```
Comment (kderusso) [inline server/src/main/java/org/elasticsearch/action/synonyms/PutSynonymRuleAction.java:67]: Should the content we're returning in the log message be truncated here if it's too long?

File: server/src/main/java/org/elasticsearch/action/synonyms/PutSynonymsAction.java [hunk 3]
```diff
@@ -61,7 +62,11 @@ public Request(StreamInput in) throws IOException {
 
         public Request(String synonymsSetId, BytesReference content, XContentType contentType) throws IOException {
             this.synonymsSetId = synonymsSetId;
-            this.synonymRules = PARSER.apply(XContentHelper.createParser(XContentParserConfiguration.EMPTY, content, contentType), null);
+            try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, content, contentType)) {
+                this.synonymRules = PARSER.apply(parser, null);
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Failed to parse: " + content.utf8ToString(), e);
```
Comment (kderusso) [inline server/src/main/java/org/elasticsearch/action/synonyms/PutSynonymsAction.java:68]: Should the content we're returning in the log message be truncated here if it's too long?

File: server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java [hunk 4]
```diff
@@ -311,26 +311,26 @@ public void getSynonymRule(String synonymSetId, String synonymRuleId, ActionList
         );
     }
 
-    public void deleteSynonymRule(String synonymSetId, String synonymRuleId, ActionListener<SynonymsReloadResult> listener) {
-        client.prepareDelete(SYNONYMS_ALIAS_NAME, internalSynonymRuleId(synonymSetId, synonymRuleId))
+    public void deleteSynonymRule(String synonymsSetId, String synonymRuleId, ActionListener<SynonymsReloadResult> listener) {
```
Comment (kderusso) [inline server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java:314]: Nice catch on the rename 😁

**Review by kderusso — APPROVED** [review]
LGTM

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml (hunk 1 above)
Comment (carlosdelest) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml:8]: Yes, this is already included in the `setup`. Not related to the PR, just housekeeping here 😊

**Pull request merged**